lib/token_authenticate_me/controllers/token_authenticateable.rb

```diff
@@ -34,6 +30,16 @@ def authenticate_token
         end
       end
 
+      def authenticate_param
+        token = params[:authentication_token]
+        session = Session.find_by_key(token)
+        @session ||= if session && (session.expiration > DateTime.now)
+                       session
+                     else
+                       false
+                     end
```

Comment:
Hmm... now that I look at this, I see that there is both instance variables and class instance variables, which is a little confusing. Why are we `@session ||= ...` here?